src/gui/src/clockWidget.cpp

```diff
@@ -152,6 +152,10 @@ void ClockTreeRenderer::resetTree()
 
 void ClockTreeRenderer::setMaxColorDepth(int depth)
 {
+  if (max_depth_ < 1) {
+    return;
+  }
+
   max_depth_ = depth;
```

Comment:
When would max_depth_ be less than 1?  If that did occur it would be unchangeable.  Did you mean `if (depth < 1)` ?